tests/update/tests.py

```diff
@@ -138,3 +140,49 @@ def test_update_respects_to_field(self):
         self.assertEqual(bar_qs[0].foo_id, a_foo.target)
         bar_qs.update(foo=b_foo)
         self.assertEqual(bar_qs[0].foo_id, b_foo.target)
+
+    def test_update_annotated_queryset(self):
+        """
+        Update of a queryset which has been annotated.
+        """
+        # Trivial annotated update first
+        qs = DataPoint.objects.annotate(alias=F('value'))
+        updated = qs.update(another_value='foo')
+        self.assertEqual(updated, 3)
```

Comment:
I wouldn't use so many intermediate variables, e.g. `self.assertEqual(qs.update(another_value='foo'), 3)` seems fine.
Also, you might include the tests that already pass on master in a separate commit so it's more clear exactly what's fixed.